core/trino-spi/src/main/java/io/trino/spi/connector/ConnectorMetadata.java

```diff
@@ -353,6 +353,16 @@ default void createTable(ConnectorSession session, ConnectorTableMetadata tableM
         throw new TrinoException(NOT_SUPPORTED, "This connector does not support creating tables");
     }
 
+    /**
+     * Drops the specified table
+     *
+     * @throws RuntimeException if the table cannot be dropped
+     */
+    default void dropTable(ConnectorSession session, SchemaTableName schemaTableName)
+    {
+        dropTable(session, getTableHandle(session, schemaTableName));
+    }
+
```

Comment:
nit: the old `dropTable` (below) should get deprecated when the new one is introduced